services/learn-card-network/brain-service/src/helpers/connection.helpers.ts

```diff
@@ -302,3 +302,127 @@ export const getConnectionStatus = async (
+        }),
+        target.findRelationships({
+            alias: 'blocked',
+            where: { relationship: {}, target: { profileId: source.profileId } },
+        }),
+    ]);
+
+    return sourceBlockedTarget.length > 0 || targetBlockedSource.length > 0;
+};
+
+/** Blocks a profile */
+export const blockProfile = async (
+    source: ProfileInstance,
+    target: ProfileInstance
+): Promise<boolean> => {
+    await Promise.all([
+        Profile.deleteRelationships({
+            alias: 'connectedWith',
+            where: {
+                source: { profileId: source.profileId },
+                target: { profileId: target.profileId },
+            },
+        }),
+        Profile.deleteRelationships({
+            alias: 'connectedWith',
+            where: {
+                source: { profileId: target.profileId },
+                target: { profileId: source.profileId },
+            },
+        }),
+        Profile.deleteRelationships({
+            alias: 'connectionRequested',
+            where: {
+                source: { profileId: source.profileId },
+                target: { profileId: target.profileId },
+            },
+        }),
+        Profile.deleteRelationships({
+            alias: 'connectionRequested',
+            where: {
+                source: { profileId: target.profileId },
+                target: { profileId: source.profileId },
+            },
+        }),
+        source.relateTo({ alias: 'blocked', where: { profileId: target.profileId } }),
+    ]);
+
+    return true;
+};
+
+/** Unblocks a profile */
+export const unblockProfile = async (
+    source: ProfileInstance,
+    target: ProfileInstance,
+    validate = true
+): Promise<boolean> => {
+    if (validate && !(await isProfileBlocked(source, target))) {
+        throw new TRPCError({
+            code: 'NOT_FOUND',
+            message: 'Profile is not blocked!',
+        });
+    }
+
+    await Profile.deleteRelationships({
+        alias: 'blocked',
+        where: {
+            source: { profileId: source.profileId },
+            target: { profileId: target.profileId },
+        },
+    });
+
+    return true;
+};
+
+export const getBlockedProfiles = async (profile: ProfileInstance): Promise<ProfileInstance[]> => {
+    return (await profile.findRelationships({ alias: 'blocked' })).map(result => result.target);
+};
+
+export const getBlockedAndBlockedByRelationships = async (
+    profile: ProfileInstance
+): Promise<ProfileInstance[]> => {
+    const [blocked, blockedBy] = await Promise.all([
+        profile.findRelationships({ alias: 'blocked' }),
+        Profile.findRelationships({
+            alias: 'blocked',
+            where: { target: { profileId: profile.profileId } },
+        }),
+    ]);
+    return [...blocked, ...blockedBy].reduce((prev, relationship): ProfileInstance[] => {
+        return [...prev, relationship.target, relationship.source];
+    }, [] as ProfileInstance[]);
```

Comment:
```suggestion

    return [...blocked, ...blockedBy].reduce<ProfileInstance[]>((profiles, relationship) => {
        profiles.push(relationship.target, relationship.source);

        return profiles;
    }, []);
```

nit: Reduce is one of the few places where you can get into a huge perf bottleneck from instantiating all those arrays! Likely wouldn't ever happen, but imagine someone blocks/is blocked by a million profiles!

Reply:
@TaylorBeeston ooh good catch, that's good to know - I remember this insight from a previous convo over in Welibrary repo now, thanks 💪 

Also thanks for the TS fix too 😎

Reply:
No problem! That one in particular is just burned into my brain from an article I read a while back where they solved nightmare performance issues caused by that 😂